src/cmd/cli/command/compose.go

```diff
@@ -63,17 +91,32 @@ func makeComposeUpCmd() *cobra.Command {
 			since := time.Now()
 			loader := configureLoader(cmd)
 
-			provider, err := getProvider(cmd.Context(), loader)
+			ctx := cmd.Context()
+			provider, err := getProvider(ctx, loader)
 			if err != nil {
 				return err
 			}
 
-			project, err := loader.LoadProject(cmd.Context())
-			if err != nil {
+			project, loadErr := loader.LoadProject(ctx)
+			if loadErr != nil {
+				term.Warn(loadErr)
+				project, err := createCustomProject(ctx, loader)
+				if err != nil {
+					return err
+				}
+
+				if !nonInteractive {
```

Comment:
No need to call `createCustomProject` when we're running non-interactively

Reply:
Could we also invert these branches to avoid the double negative?
```go
				if nonInteractive {
					return err
				}

				// ...
```